### PR TITLE
Updated Logging for document-generator

### PR DESCRIPTION
**Pull request by Mathsh**
Updated logging details in both document-generator-accounts and document-generator-api to produce more detailed information in the LOG output

Resolves SFA 722

### CODE REVIEW

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java [hunk 1]
```diff
@@ -47,7 +47,8 @@ public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequ
         try {
             transaction = transactionService.getTransaction(resourceId);
         } catch (ServiceException e) {
-            LOG.error(e, debugMap);
+            LOG.errorContext("An error occurred when calling the transaction service with resource id: "
```
Comment (jsimons1989) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java]: The first argument is the context:
```
public void errorContext(String context, Exception e, Map<String, Object> data)
```

This method signature would support the arguments:
```
public void errorContext(String context, String message, Exception e, Map<String, Object> data)
```

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
 
     @Override
     public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequest) throws DocumentInfoException {
-        LOG.info("Started getting document");
+        LOG.info("Started getting document info");
```
Comment (jsimons1989) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java]: Other APIs are using an interceptor to log start and end requests (outputs the time taken for the request to complete which helps if there's a performance issue - see company accounts API for an example). This log will be output every time a document is requested and an accounts submission but it has no details in it to link it to which document is being generated.

File: document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/document/description/impl/RetrieveApiEnumerationDescriptionImpl.java [hunk 3]
```diff
@@ -31,8 +31,10 @@ public String getApiEnumerationDescription(String fileName, String identifier, S
 
         String description = "";
 
+        LOG.info("obtaining file for api enumerations with file name: " + descriptionsFile);
         try (InputStream inputStream = new FileInputStream(descriptionsFile)) {
 
+            LOG.info("The file: " + descriptionsFile + " has been found, obtaining descriptions");
```
Comment (jsimons1989) [inline document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/document/description/impl/RetrieveApiEnumerationDescriptionImpl.java]: Can you use infoContext for the info logs please? It'll be easier to link that to the errors.

File: document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java [hunk 4]
```diff
@@ -159,17 +166,19 @@ private RenderDocumentResponse renderSubmittedDocumentData(DocumentRequest docum
...
+        if (mimeType.equals(TEXT_HTML)) {
             requestData.setContentType(mimeType);
             if (documentType != null) {
                 requestData.setDocumentType(documentType);
             } else {
                 requestData.setDocumentType(mimeType);
             }
         } else {
+            createAndLogInfoMessage("error occurred while setting content and document type, as mime type: "
+                    + mimeType + " is not valid", requestId, resourceUri, requestId);
```
Comment (jsimons1989) [inline document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java]: Wrong parameters here. Should be:
```
createAndLogInfoMessage("error occurred while setting content and document type, as mime type: "
                    + mimeType + " is not valid", resourceId, resourceUri, requestId);
```

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/service/impl/AccountsServiceImpl.java [hunk 5]
```diff
@@ -46,16 +43,27 @@ public Accounts getAccounts(String resource) throws ServiceException {
...
-
-            Map<String, Object> logMap = new HashMap<>();
-            logMap.put("resource", resource);
-            LOG.error("Failed to retreive abridged accounts data: ", e);
+            LOG.errorContext(requestId,"Failed to retrieve abridged accounts data: ", e, getDebugMap(resource));
             throw new ServiceException(e.getMessage(), e.getCause());
         }
     }
+
+    /**
```
Comment (drosser) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/service/impl/AccountsServiceImpl.java]: There's no need to add a javadoc to a private method

**Pull request merged**